### PR TITLE
[im/subscribe] Enable subscription over interaction model protocol

### PR DESCRIPTION
**Pull request by erjiaqing**
#### Problem
The current "report" routine is not spec compliance.

#### Change overview

- Update templates to use interaction model APIs for sending subscribe
requests.
- Update DeviceControllerInteractionModelDelegate to handle report
messages from subscription.
- Update ChipCallback Mgr to bridge subscribe responses to existing
callbacks, and handle TLV message just as read responses.

#### Testing

- Update TestSuite for basic Subscription protocol tests, test routing
looks like follows:
  - Send subscribe request, wait for response.
  - Wait for a few seconds, and execute "kick" commands.
  - The test is expected to receive the same number of Reports as the
  number of kick commands.
  - The test will fail when not receiving enough report data from server
  in global timeout.
- Update python script test to (in cirque) include a similar test
routine as test suite.

### CODE REVIEW

File: src/app/reporting/reporting.cpp [hunk 1]
```diff
@@ -694,53 +694,6 @@ EmberStatus emAfPluginReportingRemoveEntry(uint8_t index)
     return status;
 }
 
-void emberAfReportingAttributeChangeCallback(EndpointId endpoint, ClusterId clusterId, AttributeId attributeId, uint8_t mask,
```
Comment (asakhart) [inline src/app/reporting/reporting.cpp:697]: Please do not include old code cleanup until later changes after we get this landed and only focus on the core of this change to make the delta to enable subs easier to process.

When we go to share, we should also try to upload this change on top of a merged PR from Yunhan on a branch and do code change as one commit, followed by test changes as another commit to ease looking at code delta.

Comment (erjiaqing) [reply src/app/reporting/reporting.cpp:697]: This is the only (and necessary) removal to get code compiled, or we will have two `emberAfReportingAttributeChangeCallback` and linker will be unhappy.

Comment (asakhart) [reply src/app/reporting/reporting.cpp:697]: Yes, but why do you have to replicate the ember name? I gather something is calling that function an that's why you are just replacing it so the linker picks up your version, but couldn't you name your function differently and replace the name at the call site of emberAfReportingAttributeChangeCallback? Is that call site not available to you?

File: src/darwin/Framework/CHIP/templates/clusters-tests.zapt [hunk 2]
```diff
@@ -172,40 +172,7 @@ CHIPDevice * GetPairedDevice(uint64_t deviceId)
...
-                                             maxInterval:0xffff
-                                         responseHandler:^(NSError * err, NSDictionary * values) {
-        NSLog(@"Reporting Test Cancel Reports status: %@", err);
-
-        XCTAssertEqual(err.code, 0);
-        [expectation fulfill];
-    }];
-    [self waitForExpectationsWithTimeout:kTimeoutInSeconds handler:nil];
-}
+// Spec compliant reporting Tests goes into test suite now.
```
Comment (andy31415) [inline src/darwin/Framework/CHIP/templates/clusters-tests.zapt:175]: comment probably not needed - I would just leave the delete.

**Pull request closed without merging**